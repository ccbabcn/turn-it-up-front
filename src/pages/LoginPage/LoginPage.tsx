import FormLogin from "../../components/forms/FormLogin/FormLogin";
import { PageStyles } from "../PageStyles";

const LoginPage = () => {
  return (
    <>
      <PageStyles>
        <h2>LOGIN</h2>
        <FormLogin />
      </PageStyles>
    </>
  );
};

export default LoginPage;
