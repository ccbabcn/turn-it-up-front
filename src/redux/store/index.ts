import { configureStore } from "@reduxjs/toolkit";
import uiReducer from "../features/uiSlice/uiSlice";
import userReducer from "../features/userSlice/userSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
