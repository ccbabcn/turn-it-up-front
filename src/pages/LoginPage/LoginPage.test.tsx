import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../redux/store";
import LoginPage from "./LoginPage";
import reactTestRenderer from "react-test-renderer";

describe("Given a LoginPage component", () => {
  describe("When invoked", () => {
    test("Then it should render a heading with the text 'LOGIN'", () => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <LoginPage />
          </BrowserRouter>
        </Provider>
      );

      const renderedHeading = screen.getByRole("heading", {
        name: /expectedHeading/i,
      });

      expect(renderedHeading).toBeInTheDocument();
    });
  });
  describe("When the component is rendered", () => {
    test("Then it should match it's snapshot", () => {
      const loginPageSnap = reactTestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <LoginPage />
          </BrowserRouter>
        </Provider>
      );

      expect(loginPageSnap).toMatchSnapshot();
    });
  });
});
