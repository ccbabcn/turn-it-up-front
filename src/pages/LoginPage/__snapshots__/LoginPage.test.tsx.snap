// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginPage component When the component is rendered Then it should match it's snapshot 1`] = `
Array [
  <div
    className="sc-dkzDqf chCzKC"
  >
    <h1>
      <a
        className=""
        href="/projects"
        onClick={[Function]}
      >
        <img
          alt="Turn it Up"
          height={100}
          src="/images/logo.png"
          width={178}
        />
      </a>
    </h1>
  </div>,
  <div
    className="sc-hKMtZM kUZWga"
  >
    <h2>
      LOGIN
    </h2>
    <div
      className="sc-gsnTZi joNBmd"
    >
      <form
        autoComplete="off"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="formField"
        >
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            id="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
        <div
          className="formField"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            id="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <button
          className="form-button"
          disabled={false}
          type="submit"
        >
          LOGIN
        </button>
        <div
          className="form-options"
        >
          Don’t have an account?
           
          <a
            className=""
            href="/register"
            onClick={[Function]}
          >
            Register now
          </a>
        </div>
      </form>
    </div>
  </div>,
  <footer
    className="sc-bczRLJ VwBwN footer"
  >
    <div
      className="footer__container"
    >
      <span>
        Copyright ©2022 more info about turn it up created by cristian bermúdez
      </span>
      <div
        className="footer__row"
      >
        <div
          className="footer__col"
        >
          <ul>
            <li
              className="title"
            >
              About
            </li>
            <li>
              The Web
            </li>
            <li>
              More Info
            </li>
            <li>
              Coming soon
            </li>
          </ul>
        </div>
        <div
          className="footer__col"
        >
          <ul>
            <li
              className="title"
            >
              Sponsors
            </li>
            <li>
              Music and more
            </li>
            <li>
              Theme Store
            </li>
            <li>
              Rednef
            </li>
          </ul>
        </div>
        <div
          className="footer__col"
        >
          <ul>
            <li
              className="title"
            >
              FAQs
            </li>
            <li>
              What project
            </li>
            <li>
              Which role
            </li>
            <li>
              More
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>,
]
`;
